src/utils/hooks/api/use-get-prices.tsx

```diff
@@ -36,25 +38,87 @@ const composeUrl = (assetsIds: string): string => {
   return url.toString();
 };
 
-const getPricesByTicker = (currencies: CurrencyExt[], prices: Prices) =>
-  currencies.reduce((acc, currency) => {
+const fetchAPI = async (currencies: CurrencyExt[]): Promise<Prices> => {
+  const assetsIds = composeIds(currencies);
+  const url = composeUrl(assetsIds);
+
+  const response = await fetch(url);
+  return response.json();
+};
+
+/**
+ * Fetched prices from an API (i.e. CoinGecko)
+ * @param {CurrencyExt[]} currencies Currencies to be fetched.
+ * @return {Promise<Prices>} Prices groupped by currency tickers
+ */
+const getCoinGeckoPrices = async (currencies: CurrencyExt[]): Promise<Prices> => {
+  const data = await fetchAPI(currencies);
+
+  return currencies.reduce((acc, currency) => {
     const coingeckoId = getCoingeckoId(currency);
-    return { ...acc, [currency.ticker]: prices[coingeckoId] };
+    return { ...acc, [currency.ticker]: data[coingeckoId] };
   }, {});
+};
+
+const baseBTCAmount = newMonetaryAmount(1, Bitcoin, true);
+
+const calculateCurrencyPrice = (btcPriceUSD: Big, exchangeRate: ExchangeRate<Bitcoin, CurrencyExt>): Big => {
+  const currencyExchangeRate = exchangeRate.toCounter(baseBTCAmount).toBig();
+  return btcPriceUSD.div(currencyExchangeRate);
+};
 
-const getPrices = async (currencies?: CurrencyExt[]): Promise<Prices | undefined> => {
-  if (!currencies) {
-    return;
+const getOracleCurrencyPrice = async (btcPriceUSD: Big, currency: CurrencyExt): Promise<Big> => {
+  if (currency.ticker === WRAPPED_TOKEN.ticker) {
+    return btcPriceUSD;
   }
 
-  const allCurrencies = [Bitcoin, ...currencies];
-  const assetsIds = composeIds(allCurrencies);
-  const url = composeUrl(assetsIds);
+  try {
+    const exchangeRate = await window.bridge.oracle.getExchangeRate(currency);
 
-  const response = await fetch(url);
-  const pricesByCoingeckoId = await response.json();
+    return calculateCurrencyPrice(btcPriceUSD, exchangeRate);
+  } catch (e) {
+    return new Big(0);
+  }
+};
 
-  return getPricesByTicker(allCurrencies, pricesByCoingeckoId);
+/**
+ * Fetched prices from an oracle are based on exchange rates between the currency
+ * and BTC (i.e USDT/BTC). That's why only the price of bitcoin is fetched from
+ * external api (i.e CoinGecko). By using the exchange rate between a currency and
+ * BTC and the current BTC price, we are able to calculate the price of that currency.
+ *
+ * @param {CurrencyExt[]} currencies Currencies to be fetched.
+ * @return {Promise<Prices>} Prices groupped by currency tickers
+ */
+const getOraclePrices = async (currencies: CurrencyExt[]): Promise<Prices> => {
+  const data = await fetchAPI([Bitcoin]);
```

Comment:
> Member

@nud3l, so if I fully understood your feedback, we might need to export this out of `use-get-prices` (probably a new hook called `use-get-exchange-rates`), because we won't translate these exchanges rates into `USD`, but just use them for the calculation. Correct?

Reply:
Well, you can still keep in in the same API but you would get USD values when you use coingecko and BTC values when you use the oracle.

Either way is viable and depends on how you want to organize the code. But for that I would refer to @tomjeatt

Reply:
AMM first. As lending is just on testnet, let's get the AMM out first.

Thanks!